pkg/pod/pod.go

```diff
@@ -91,10 +91,14 @@ type Builder struct {
 	OverrideHomeEnv bool
 }
 
+// Transformer is a function that will transform a Pod. This can be used to mutate
+// a Pod generated by Tekton after it got generated.
+type Transformer func(*corev1.Pod) (*corev1.Pod, error)
```

Comment:
It looks like this is an interface with exactly one implementation, which smells a bit. Are future transformers planned?

The variadic transformers below also looks like a way to add behavior to `Build` without having to update all callers, which is clever, but also a bit smelly. Maybe `Builder.SetLimitRangeTransformer(limitrange.Transformer)` would accomplish the same thing, a bit cleaner?

Reply:
Ah I see https://github.com/tektoncd/pipeline/pull/4176#discussion_r693010994 now -- makes sense, ignore this.